elytron/src/main/java/org/wildfly/extension/elytron/ElytronPersistentResourceXMLDescriptionFactory.java

```diff
@@ -0,0 +1,230 @@
+            builder.addChild(new SaslParser().parser);
+        }
+
+        addTlsParser(schema, builder);
+        addCredentialStoreParser(schema, builder);
+
+        if (schema.since(ElytronSubsystemSchema.VERSION_13_0)) {
+            builder.addChild(getExpressionResolverParser());
+        }
+
+        if (schema.since(ElytronSubsystemSchema.VERSION_3_0)) {
+            builder.addChild(getDirContextParser());
+        }
+
+        addPolicyParser(schema, builder);
+        if (schema.since(ElytronSubsystemSchema.VERSION_5_0)) {
+            builder.addChild(new JaspiConfigurationParser().jaspiConfigurationParser_5_0);
+        }
+
+        return builder.build();
+    }
+
+    private static void addPolicyParser(ElytronSubsystemSchema schema, PersistentResourceXMLDescription.PersistentResourceXMLBuilder builder) {
+        PolicyParser policyParser = new PolicyParser();
+        if (schema.since(ElytronSubsystemSchema.VERSION_1_2)) {
+            builder.addChild(policyParser.parser_1_2);
+        } else if (schema.since(ElytronSubsystemSchema.VERSION_1_1)) {
+            builder.addChild(policyParser.parser_1_0);
+        }
+    }
+
+    private static void addCredentialStoreParser(ElytronSubsystemSchema schema, PersistentResourceXMLDescription.PersistentResourceXMLBuilder builder) {
+        CredentialStoreParser credentialStoreParser = new CredentialStoreParser();
+        if (schema.since(ElytronSubsystemSchema.VERSION_13_0)) {
+            builder.addChild(credentialStoreParser.getCredentialStoresParser_13().build());
+        } else if (schema.since(ElytronSubsystemSchema.VERSION_1_0)) {
+            builder.addChild(credentialStoreParser.getCredentialStoresParser().build());
+        }
+    }
+
+    private static void addTlsParser(ElytronSubsystemSchema schema, PersistentResourceXMLDescription.PersistentResourceXMLBuilder builder) {
+        TlsParser tlsParser = new TlsParser();
+        if (schema.since(ElytronSubsystemSchema.VERSION_18_0_COMMUNITY)) {
+            builder.addChild(tlsParser.tlsParserCommunity_18_0);
+        } else if (schema.since(ElytronSubsystemSchema.VERSION_14_0)) {
+            builder.addChild(tlsParser.tlsParser_14_0);
+        } else if (schema.since(ElytronSubsystemSchema.VERSION_12_0)) {
+            builder.addChild(tlsParser.tlsParser_12_0);
+        } else if (schema.since(ElytronSubsystemSchema.VERSION_9_0)) {
+            builder.addChild(tlsParser.tlsParser_9_0);
+        } else if (schema.since(ElytronSubsystemSchema.VERSION_8_0)) {
+            builder.addChild(tlsParser.tlsParser_8_0);
+        } else if (schema.since(ElytronSubsystemSchema.VERSION_5_0)) {
+            builder.addChild(tlsParser.tlsParser_5_0);
+        } else if (schema.since(ElytronSubsystemSchema.VERSION_4_0)) {
+            builder.addChild(tlsParser.tlsParser_4_0);
+        } else if (schema.since(ElytronSubsystemSchema.VERSION_1_0)) {
+            builder.addChild(tlsParser.tlsParser_1_0);
+        }
```

Comment:
These static PersistenceResourceXMLDescription references kind of defeat the purpose of paramaterizing your parser.  Would it not be better to build this PersistenceResourceXMLDescription in one place?  That way you can easily tell, at a glance, what changed when.

Reply:
@pferraro Do you mean not having separate parsers for TLS, credential store etc. but registering everything at one place based on version?

Reply:
or  maybe that is not what you meant. What did you mean by building PersistenceResourceXMLDescription in one place? As it is being built in this apply method at one place IIUC